tests/SCsub

```diff
@@ -13,3 +13,9 @@ env_tests.add_source_files(env.tests_sources, "*.cpp")
 
 lib = env_tests.add_library("tests", env.tests_sources)
 env.Prepend(LIBS=[lib])
+
+# we must disable the THREAD_LOCAL entirely in doctest to prevent crashes on debugging
+# since we link with /MT thread_local is always expired when the header is used, so the debugger crashes the engine and it causes weird errors
+# explained in https://github.com/onqtam/doctest/issues/401
+if env_tests["platform"] == "windows":
+    env_tests.Append(CPPDEFINES=["DOCTEST_THREAD_LOCAL"])
```

Comment:
```suggestion
    env_tests.Append(CPPDEFINES={"DOCTEST_THREAD_LOCAL" : ""})
```

Compiles on Windows, I think this is the way.

Reply:
Or this:
```suggestion
    env_tests.Append(CPPDEFINES=[("DOCTEST_THREAD_LOCAL", "")])
```
As used elsewhere:
```
platform/windows/detect.py
394:    env.Append(CPPDEFINES=["WINDOWS_ENABLED", "WASAPI_ENABLED", "WINMIDI_ENABLED"])

platform/uwp/detect.py
155:    env.Append(CPPDEFINES=[("WINVER", winver), ("_WIN32_WINNT", winver), "WIN32"])
```

Reply:
I ended up with ("DOCTEST_THREAD_LOCAL", "")